torch/onnx/symbolic_opset9.py

```diff
@@ -4586,9 +4586,9 @@ def try_mask_to_index(index):
             rank = symbolic_helper._get_tensor_rank(self)
             if rank is None:
                 raise NotImplementedError(
```

Comment:
For your consideration: we can also make use of the new errors.SymbolicValueError to provide more context:

https://github.com/pytorch/pytorch/blob/ba90c9f2298433778cc6a7a2008d0299aa2911da/torch/onnx/errors.py#L55

Reply:
I am not sure if I followed your suggestion, as this PR did not introduce any new `raise` nor updated any.

https://github.com/pytorch/pytorch/pull/83007/ introduced "OnnxExporterError", "CheckerError", "UnsupportedOperatorError" and "SymbolicValueError" exceptions, but it didn't replace all occurrences of the deprecated ones, leading to the mix we are seeing now.

Changing just this instance would not help, as there are currently 105 old fashioned exceptions within `torch/onnx` folder (37 in this file). There are only 16 new style exceptions being used. 

Given this, I'd suggest creating a separate PR just for refactoring the deprecated exceptions.
In fact, for future refactoring, I would also recommend replacing the old style when a new version is added to prevent unnecessary noise on PR code and code review.

By the way, having ONNX-exporter specific exceptions is a great improvement. I believe there was a task for this in one of our dashboards :)

Reply:
Makes sense! I can create a follow up